apis/apiserver/v1/vshn_keycloak_backup_types.go

```diff
@@ -0,0 +1,74 @@
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"sigs.k8s.io/apiserver-runtime/pkg/builder/resource"
+)
+
+// VSHNKeycloakBackup needs to implement the builder resource interface
+var _ resource.Object = &VSHNKeycloakBackup{}
+var _ resource.ObjectList = &VSHNKeycloakBackupList{}
+
+// +kubebuilder:object:root=true
+// +k8s:openapi-gen=true
+type VSHNKeycloakBackup struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Status VSHNKeycloakBackupStatus `json:"status,omitempty"`
+}
+
+// +k8s:openapi-gen=true
+type VSHNKeycloakBackupStatus struct {
+	// DatabaseBackupAvailable indicates if this backup contains a database backup for Keycloak.
+	// Not every file backup might have a database backup associated, because the retention is not enforced at the same time.
```

Comment:
Does this comment actually apply here? Or rather the whole thing? There's only a database backup for Keycloak, unlike Nextcloud where we have files and database.

Reply:
yep this seems to be taken from Nextcloud

Reply:
Ah yes, classic copy-pasta 🍝